frontend/libs/studio-components/src/components/StudioTextField/StudioTextField.tsx

```diff
@@ -0,0 +1,22 @@
+import React, { useState } from 'react';
+import type { StudioTextfieldToggleViewProps } from './StudioTextfieldToggleView/StudioTextfieldToggleView';
+import {
+  StudioIconTextfield,
+  type StudioIconTextfieldProps,
+} from './StudioIconTextfield/StudioIconTextfield';
+import { StudioTextfieldToggleView } from './StudioTextfieldToggleView/StudioTextfieldToggleView';
+
+export type StudioTextFieldProps = {
+  viewProps: StudioTextfieldToggleViewProps;
+  inputProps: StudioIconTextfieldProps;
+};
+export const StudioTextField = ({ inputProps, viewProps }: StudioTextFieldProps) => {
+  const [isViewMode, setIsViewMode] = useState(true);
+
+  const toggleViewMode = () => {
+    setIsViewMode((prevMode) => !prevMode);
+  };
+
+  if (isViewMode) return <StudioTextfieldToggleView onClick={toggleViewMode} {...viewProps} />;
+  return <StudioIconTextfield onBlur={toggleViewMode} {...inputProps} />;
```

Comment:
I suggest adding f.ex. the `autoFocus` prop here to improve the `onBlur` functionality. Since the text field is in focus, the action of clicking outside the text field will always call the onBlur-function, making the experience more seamless for the user.

```suggestion
  return <StudioIconTextfield onBlur={toggleViewMode} {...inputProps} autoFocus />;
```